src/cid.js

```diff
@@ -86,20 +89,11 @@ export class CID {
 
     // Circular reference
     /** @readonly */
-    this.asCID = this
-
-    // Configure private properties
-    Object.defineProperties(this, {
-      byteOffset: hidden,
-      byteLength: hidden,
-
-      code: readonly,
-      version: readonly,
-      multihash: readonly,
-      bytes: readonly,
+    this.#asCID = this
+  }
 
-      asCID: hidden
-    })
+  get asCID () {
```

Comment:
oh, this is making it public again? so this is essentially a hack to make a hidden property? sneaky.

Reply:
I don't think this is going to survive the structural cloning which is what going to occur when transferring CIDs across realms/threads.

If it does not it would defeat the purpose

Reply:
@Gozala are there any tests that guard against regressions around this?  I can't see anything obvious.

Reply:
We should have, but we did not as far as I can tell. I have created one
https://github.com/multiformats/js-multiformats/pull/206/files